### PR TITLE
Register plug-in as Packer integration

### PR DESCRIPTION
**Pull request by nywilken**
:wave: fellow Packer maintainer working on migrating your plugin to the Packer integration framework.

This change takes the necessary steps to register this plugin as an official Packer integration.
Integrations can be found on the Packer integration portal at https://developer.hashicorp.com/packer/integrations.

---

The pull-request consists of the following changes:

- Adds controlling, metadata file, `metadata.hcl` for registering the plug-in and it components as integrations.
Details on the contents, along with a description of the attributes, can be found at https://github.com/hashicorp/integration-template.
- Adds the GitHub action workflows for triggering manual and automatic integration updates.
- Restructures the plug-in documentation to match the expected format of the integration framework.
- Adds a `.web-docs` directory for serving the fully render documentation as the integration docs.
- Adds the build-docs make target `make build-docs` for syncing changes to the `docs` directory to the `.web-docs` directory.

Changes to the integration docs can be made at plugin release via the `notify-integration-release-via-tag` workflow or
manually by running the `notify-integration-release-via-manual` workflow.

Details on how the Integration framework pipeline works can be found at https://github.com/hashicorp/packer/pull/12702

### TODOs
- [x] Open pull-request against external plugin.
- [ ] Update integration description [`.web-docs/metadata.hcl`](.web-docs/metadata.hcl).
- [ ] Packer team open internal pull-request to enable integration.
- [ ] Review plugin integration on Packer integration portal .... Iterate

### CODE REVIEW

File: .github/workflows/notify-integration-release-via-manual.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          if [[ -z "$(git status -s)" ]]; then
+            echo "OK"
+          else
+            echo "Docs have been updated, but the compiled docs have not been committed."
+            echo "Run 'make generate', and commit the result to resolve this error."
+            exit 1
+          fi
+      # Perform the Release
+      - name: Checkout integration-release-action
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
```
Comment (zliang-akamai) [inline .github/workflows/notify-integration-release-via-manual.yaml:38]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/notify-integration-release-via-manual.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      branch:
+        description: "A branch or SHA"
+        default: 'main'
+        required: false
+jobs:
+  notify-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout this repo
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
```
Comment (zliang-akamai) [inline .github/workflows/notify-integration-release-via-manual.yaml:20]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/go-test-multiplatform.yml [hunk 3]
```diff
@@ -28,25 +28,26 @@ jobs:
...
+  get-test-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install system deps
+        run: sudo apt-get install -y build-essential
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
```
[the remark this answers is not in the document]
Comment (zliang-akamai) [reply .github/workflows/go-test-multiplatform.yml:41]: Actually, this is causing the CI failures. I think we can revert changes in this file and we will look into this workflow and fix it later.

[the remark this answers is not in the document]
Comment (zliang-akamai) [reply .github/workflows/go-test-multiplatform.yml:41]: https://github.com/linode/packer-plugin-linode/pull/161

**Review by lgarber-akamai — APPROVED** [review]
LGTM, thanks for the contribution!

**Review by zliang-akamai — APPROVED** [review]

File: .github/workflows/notify-integration-release-via-tag.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          REF: ${{ github.ref_name }}
+        run: |
+          echo "packer-version=$(echo "$REF" | sed -E 's/v?([0-9]+\.[0-9]+\.[0-9]+)/\1/')" >> "$GITHUB_OUTPUT"
+  notify-release:
+    needs:
+      - strip-version
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout this repo
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
```
Comment (zliang-akamai) [inline .github/workflows/notify-integration-release-via-tag.yaml:24]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/notify-integration-release-via-tag.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        with:
+          ref: ${{ github.ref }}
+      # Ensure that Docs are Compiled
+      - uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
```
Comment (zliang-akamai) [inline .github/workflows/notify-integration-release-via-tag.yaml:28]:
```suggestion
      - uses: actions/setup-go@v4
```

File: .github/workflows/notify-integration-release-via-tag.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+          if [[ -z "$(git status -s)" ]]; then
+            echo "OK"
+          else
+            echo "Docs have been updated, but the compiled docs have not been committed."
+            echo "Run 'make generate', and commit the result to resolve this error."
+            exit 1
+          fi
+      # Perform the Release
+      - name: Checkout integration-release-action
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
```
Comment (zliang-akamai) [inline .github/workflows/notify-integration-release-via-tag.yaml:42]:
```suggestion
        uses: actions/checkout@v4
```

**Review by zliang-akamai — APPROVED** [review]

**Pull request merged**